src/app/pages/masterdata/department.functionality/department.functionality.component.ts

```diff
@@ -101,14 +113,64 @@ export class DepartmentFunctionalityComponent implements OnInit {
                         }
                     }
                 }
+                this.checkAllStatusView();
+                this.checkAllStatusOnlyHOD();
+            });
+    }
+     selectAllDeptView(value: any) : void{
+        if(value.checked)
+        {
+            this.pagesForDepartment.forEach(x =>{
+                x.AllowView = true;
+            });
+        }
+        else{
+            this.pagesForDepartment.forEach(x =>{
+                x.AllowView = false;
+            });
+        }
+
+    }
+   selectAllDeptOnlyHOD(value: any) : void{
+        if(value.checked)
+        {
+            this.pagesForDepartment.forEach(x =>{
+                x.OnlyForHod = true;
```

Comment:
Instead of using true or false separately you can use "value.checked" directly in "xOnlyForHod" inside the forEach loop,
Please remove the If block